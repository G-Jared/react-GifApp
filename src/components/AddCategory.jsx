import { useState } from "react";


export const AddCategory = ({onNewCategory}) => {//Agregas el parametro de las props setCategories en este caso

    const [inputValue, setinputValue] = useState('');//Al mandar valor inicial debe usarse onChange
    const onInputChange = ({target})=>{
        setinputValue( target.value );
    }

    const onSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length <=1) return; 

        //setCategories( categories=>[inputValue, ...categories] );
        //desestructuras categories y agregas el inputValue al arreglo inicial

        onNewCategory( inputValue.trim() );
        setinputValue('');
    }


  return (
    <form onSubmit={onSubmit} >
        <input 
            type="text"
            placeholder="Buscar Gifs" 
            value={inputValue}
            onChange={onInputChange}

        />
    </form>
  )
}
