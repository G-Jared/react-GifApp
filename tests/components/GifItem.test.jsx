import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Prubebas en <GifItem />', () => {

    const title = 'DBZ';
    const url = 'https://dbz.com/dbz.jpg';

    test('Hacer match con snapshot', () => {

        const { container } = render( <GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar la imagen con el url y el alt indicado', () => {

        render( <GifItem title={ title } url={ url } />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( title );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });

    test('Comprobar que el titulo se muestre en el componente', () => {

        render( <GifItem title={ title } url={ url } />);
        expect( screen.getByText( title ) ).toBeTruthy();
        
    });
});