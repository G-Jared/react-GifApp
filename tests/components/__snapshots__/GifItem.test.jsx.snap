// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubebas en <GifItem /> Hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="DBZ"
      src="https://dbz.com/dbz.jpg"
    />
    <p>
      DBZ
    </p>
  </div>
</div>
`;
